components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,463 @@
+
+		currentPlace = placement.placeEnd
+		table.insert(self.placements, placement)
+	end
+
+	return self.placements
+end
+
+function PrizePool:setConfig(option, value)
+	self.options[option] = value
+	return self
+end
+
+function PrizePool:addCustomConfig(name, default, func)
+	config[name] = {
+		default = default,
+		func = func
+	}
+	return self
+end
+
+-- TODO extract parts of this to the future Enum class
+local CUSTOM_TYPES_OFFSET = 100
+local CUSTOM_TYPES_USED = 0
+function PrizePool:addCustomPrizeType(enum, data)
+	prizeTypes[enum] = CUSTOM_TYPES_OFFSET + CUSTOM_TYPES_USED
+	CUSTOM_TYPES_USED = CUSTOM_TYPES_USED + 1
+	prizeData[prizeTypes[enum]] = data
+	return self
+end
+
+function PrizePool:addPrize(enum, index, data)
+	assert(prizeTypes[enum], 'addPrize: Not a valid prizeEnum!')
+	assert(Logic.isNumeric(index), 'addPrize: Index is not numeric!')
+	table.insert(self.prizes, {id = enum .. index, enum = prizeTypes[enum], index = index, data = data})
+	return self
+end
+
+function PrizePool:setWidgetInjector(widgetInjector)
+	assert(widgetInjector:is_a(WidgetInjector), "setWidgetInjector: Not a WidgetIjector")
+	self.widgetInjector = widgetInjector
+	return self
+end
+
+function PrizePool:setLpdbInjector(lpdbInjector)
+	-- TODO: Add type check once interface has been created
+	self.lpdbInjector = lpdbInjector
+	return self
+end
+
+function PrizePool:_hasUsdPrizePool()
+	return (Table.any(self.placements, function (_, placement)
+		return placement.hasUSD
+	end)) or (self.options.autoUSD and Table.any(self.prizes, function (_, prize)
+		return prize.enum == prizeTypes.LOCAL_CURRENCY
+	end))
+end
+
+--
+-- Static Functions
+--
+
+--- Remove all non-numeric characters from an input and changes it to a number.
+-- Most commonly used on money inputs, as they often contain , or .
+function PrizePool._parseInteger(input)
+	return tonumber((string.gsub(input, '[^%d.]', '')))
+end
+
+--- Checks if the input matches the format of a date
+function PrizePool._isValidDateFormat(date)
+	if String.isEmpty(date) then
+		return false
+	end
+	return string.match(date, '%d%d%d%d-%d%d-%d%d') and true or false
+end
+
+--- Checks that an Opponent Struct is valid and has a valid type
+function PrizePool._assertOpponentStructType(typeStruct)
+	if not typeStruct then
+		error('Please provide a type!')
+	elseif type(typeStruct) ~= 'table' or not typeStruct.type then
+		error('Could not parse type!')
+	elseif not Opponent.isType(typeStruct.type) then
+		error('Not a valid type!')
+	end
+end
+
+--- Fetches the LPDB object of a tournament
+function PrizePool._getTournamentInfo(pageName)
+	return mw.ext.LiquipediaDB.lpdb('tournament', {
+		conditions = '[[pagename::' .. pageName .. ']]',
+		limit = 1,
+	})[1]
+end
+
+--- Get the default date based on wiki-variables set in the Infobox League
+function PrizePool._getTournamentDate()
+	return Variables.varDefaultMulti('tournament_enddate', 'tournament_edate', 'edate', TODAY)
```

Comment:
`tournament_edate` and `edate` are no longer used

Reply:
Legacy support